chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py

```diff
@@ -0,0 +1,157 @@
+            bboxes (~chainer.Variable): Batched bounding boxes.
+                Its shape is :math:`(N, R, 4)`.
+            labels (~chainer.Variable): Batched labels.
+                Its shape is :math:`(N, R)`.
+            scale (float or ~chainer.Variable): Amount of scaling applied to
+                the raw image during preprocessing.
+
+        Returns:
+            chainer.Variable:
+            Loss scalar variable.
+            This is a sum of losses for Region Proposal Network and
+            the head module.
+
+        """
+        if isinstance(bboxes, chainer.Variable):
+            bboxes = bboxes.data
+        if isinstance(labels, chainer.Variable):
+            labels = labels.data
+        if isinstance(scale, chainer.Variable):
+            scale = scale.data
+        scale = np.asscalar(cuda.to_cpu(scale))
+        n = bboxes.shape[0]
+        if n != 1:
+            raise ValueError('currently only batch size 1 is supported')
+
+        img_size = imgs.shape[2:][::-1]
+
+        features = self.faster_rcnn.extractor(imgs, test=not self.train)
+        rpn_locs, rpn_scores, rois, roi_indices, anchor = self.faster_rcnn.rpn(
+            features, img_size, scale, test=not self.train)
+
+        # Since batch size is one, convert variables to singular form
+        bbox = bboxes[0]
+        label = labels[0]
+        rpn_score = rpn_scores[0]
+        rpn_loc = rpn_locs[0]
+        roi = rois
+
+        # Sample RoIs and forward
+        sample_roi, gt_roi_cls_loc, gt_roi_label, roi_loc_in_weight, \
+            roi_loc_out_weight = self.proposal_target_creator(roi, bbox, label)
+        sample_roi_index = self.xp.zeros((len(sample_roi),), dtype=np.int32)
+        roi_cls_loc, roi_score = self.faster_rcnn.head(
+            features, sample_roi, sample_roi_index, test=not self.train)
+
+        # RPN losses
+        gt_rpn_loc, gt_rpn_label, rpn_loc_in_weight, rpn_loc_out_weight =\
+            self.anchor_target_creator(bbox, anchor, img_size)
+
+        rpn_cls_loss = F.softmax_cross_entropy(rpn_score, gt_rpn_label)
+        rpn_loc_loss = _smooth_l1_loss(
+            rpn_loc, gt_rpn_loc,
+            rpn_loc_in_weight, rpn_loc_out_weight, self.rpn_sigma)
+
+        # Losses for outputs of the head.
+        cls_loss = F.softmax_cross_entropy(roi_score, gt_roi_label)
+        loc_loss = _smooth_l1_loss(
+            roi_cls_loc, gt_roi_cls_loc,
+            roi_loc_in_weight, roi_loc_out_weight, self.sigma)
+
+        loss = rpn_loc_loss + rpn_cls_loss + loc_loss + cls_loss
+        chainer.reporter.report({'rpn_loc_loss': rpn_loc_loss,
+                                 'rpn_cls_loss': rpn_cls_loss,
+                                 'loc_loss': loc_loss,
+                                 'cls_loss': cls_loss,
+                                 'loss': loss},
+                                self)
+        return loss
+
+
+def _smooth_l1_loss(x, t, in_weight, out_weight, sigma):
```

Comment:
Thanks for your suggestion.
I looked into it, and I found that parameterization used in the Huber loss and smooth L1 are different.

I think that the current style is OK.
After merging this, I am planning to add `_bbox_loss` to `links.loss` with a proper documentation on the formula used.